projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -151,6 +151,15 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
             this.grid.validation.update(cell.id.rowID, newRowData);
         }
         if (this.grid.primaryKey === cell.column.field) {
+            if (this.grid.pinnedRecords.length > 0) {
+                const cellRow = this.grid.getRowByIndex(cell.rowIndex);
```

Comment:
why do you need to get the row component .. I believe that all the necessary info used below can be directly accessed from the cell and the edit event args like in the selection below